go/libraries/doltcore/merge/merge.go

```diff
@@ -320,84 +379,202 @@ func mergeProllyRowData(ctx context.Context, postMergeSchema, rootSch, mergeSch,
 
 var syncPool = pool.NewBuffPool()
 
-//type valueMerger struct {
-//	numCols                                int
-//	vD, lVD, rVD, bVD                      val.TupleDesc
-//	leftMapping, rightMapping, baseMapping map[int]int
-//}
-//
-//func newValueMerger(merged, leftSch, rightSch, baseSch schema.Schema) *valueMerger {
-//	n := merged.GetNonPKCols().Size()
-//	leftMapping := make(map[int]int, n)
-//	rightMapping := make(map[int]int, n)
-//	baseMapping := make(map[int]int, n)
-//
-//	for i, tag := range merged.GetNonPKCols().Tags {
-//		if j, ok := leftSch.GetNonPKCols().TagToIdx[tag]; ok {
-//			leftMapping[i] = j
-//		}
-//		if j, ok := rightSch.GetNonPKCols().TagToIdx[tag]; ok {
-//			rightMapping[i] = j
-//		}
-//		if j, ok := baseSch.GetNonPKCols().TagToIdx[tag]; ok {
-//			baseMapping[i] = j
-//		}
-//	}
-//
-//	return &valueMerger{
-//		n,
-//		prolly.ValueDescriptorFromSchema(merged),
-//		prolly.ValueDescriptorFromSchema(leftSch),
-//		prolly.ValueDescriptorFromSchema(rightSch),
-//		prolly.ValueDescriptorFromSchema(baseSch),
-//		leftMapping,
-//		rightMapping,
-//		baseMapping,
-//	}
-//}
-//
-//// tryMerge performs a cell-wise merge given left, right, and base cell value
-//// tuples. It returns the merged cell value taple and an ok bool.
-//func (m *valueMerger) tryMerge(left, right, base val.Tuple) (merged val.Tuple, ok bool) {
-//	processColumnFunc := func(i int) (resultVal []byte, isConflict bool) {
-//		var leftVal []byte
-//		if l, ok := m.leftMapping[i]; ok {
-//			leftVal = m.lVD.GetField(l, left)
-//		}
-//		var rightVal []byte
-//		if r, ok := m.rightMapping[i]; ok {
-//			rightVal = m.rVD.GetField(r, right)
-//		}
-//		var baseVal []byte
-//		if b, ok := m.baseMapping[i]; ok {
-//			baseVal = m.bVD.GetField(b, base)
-//		}
-//
-//		// bytes.Equal?? What about nil vs empty slices??
-//		leftModified := m.vD.Comparator().CompareValues(leftVal, baseVal, m.vD.Types[i]) != 0
-//		rightModified := m.vD.Comparator().CompareValues(rightVal, baseVal, m.vD.Types[i]) != 0
-//
-//		switch {
-//		case leftModified && rightModified:
-//			return nil, true
-//		case leftModified:
-//			return leftVal, false
-//		default:
-//			return rightVal, false
-//		}
-//	}
-//
-//	mergedValues := make([][]byte, m.numCols)
-//	for i := 0; i < m.numCols; i++ {
-//		v, isConflict := processColumnFunc(i)
-//		if isConflict {
-//			return nil, false
-//		}
-//		mergedValues[i] = v
-//	}
-//
-//	return val.NewTuple(syncPool, mergedValues...), true
-//}
+type valueMerger struct {
+	numCols                                int
+	vD, lVD, rVD, bVD                      val.TupleDesc
+	leftMapping, rightMapping, baseMapping map[int]int
```

Comment:
I don't think I can use `val.OrdinalMapping` because there may not be a match for column in the merged schema. For example, if right adds a column, the merged schema will contain the added column, but left will not. 

I could store `-1` in the val.OrdinalMapping if necessary and check for it.